app/indices/cms_page_index.rb

```diff
@@ -8,5 +8,5 @@
   has tenant_id, type: :bigint
 
   indexes :published
-  scope { CMS::Page.where(searchable: true) }
+  scope { CMS::Page.where(searchable: true).where.not(published: nil) }
```

Comment:
Maybe just add `return false if published.nil?` here https://github.com/3scale/porta/blob/e6cf20e237e75c303504c8375dc3819581c9971f/app/models/cms/page.rb#L68-L83

Reply:
Yeah, but I mean that `searchable`defines some criteria of when a page should be "searchable". We just add a new criteria here "the page must be searchable if it's **published**, so I just suggested to put it all to the same bucket. But that's fine as it is.

Reply:
It makes logical sense. But then we will need to ensure to change this value reliably when page is published/unpublished and I'm not really keen on making sure all details are correct. It also follows previous (bad?) example of filtering out unpublished pages separately in the presenter.

Reply:
If the would move in sync, then what is the point to have both of these? 

This `searchable` attribute doesn't make much sense to me to begin with. To make it right IMO a bigger refactoring would be needed.

And I don't find it a priority to make these attributes make sense. It requires more attention that now is due elsewhere.

For now I think this is an improvement and we can have a JIRA to sanitize these attributes.

Reply:
> If the would move in sync, then what is the point to have both of these?

I don't understand this.

Reply:
"they"

I meant, if both attributes are changed together, then why keep them. I think this `searchable` is probably redundant and we may just define the proper index scope to skip the elements currently defining it. Why keep a separate field in the database when index can have a scope?

Reply:
But again, not something I find very useful to spend time on now.

Reply:
Understood, thanks.